src/video/wayland/SDL_waylandwindow.c

```diff
@@ -1952,11 +1931,6 @@ Wayland_HandleResize(SDL_Window *window, int width, int height, float scale)
                                         0, 0);
     }
 
-    region = wl_compositor_create_region(data->waylandData->compositor);
-    wl_region_add(region, 0, 0, window->w, window->h);
-    wl_surface_set_opaque_region(data->surface, region);
-    wl_region_destroy(region);
```

Comment:
The timing of this I believe is important for the block that comes immediately after it - I would check the repro case described in #4563 before changing this.

Reply:
Nah, opaque region should be completely irrelevant to that issue. It's just a performance hint for the compositor and the protocol requires it to gracefully handle incorrect values (but on an unrelated note - SDL windows can be set to have alpha, right? so this probably should be conditional... :grinning: )

Reply:
Good enough for me! (For alpha I'm not sure, both EGL and Vulkan WSI don't seem to care too much about the "opaque" part of opaque_region... I just figured it was a different word for "visible" in this case)

Reply:
It's the compositor who has to care, and with opaque_region it's free to optimize stuff away (like not drawing other windows that are obscured by it at all). I'm not sure whether any of them actually does, but even if that's true right now it will likely change in the future (and I can already say that we sure want to use it in phoc at some point).

Reply:
I think that Mutter checks the opaque region as a prerequisite for unredirection, so setting it is rather important with fullscreen windows.

Checking the window opacity value and only setting the opaque region if the window transparency is equal to 1.0 is trivial though, so no reason not to do it.

Reply:
Wait, what does `window->opacity` even do here? Is it relevant to Wayland backend at all? From a quick glance I don't think it is (and frankly I don't really see a way for it to be).

What I meant is that SDL should not set opaque region when `SDL_HINT_VIDEO_EGL_ALLOW_TRANSPARENCY` is set.

Reply:
Ah, I didn't know that hint existed.  Fixed.

In SDL3 there should probably be a per-window transparency flag to set whether or not the window should be opaque as a global hint is a bit of a blunt instrument, but for now, it is what it is.